src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -101,14 +101,25 @@
      */
     private /*almost final*/ transient CpsFlowExecution execution;
 
+    /**
+     * @deprecated use {@link #runtimeThreads}
+     */
+    @Deprecated
+    private final Map<Integer, CpsThread> threads = new HashMap<>();
```

Comment:
If adopting my `writeReplace` suggestion:

```suggestion
    private final Map<Integer, CpsThread> threads;
```

Reply:
Close, but `final` also has to be removed for this to compile with your other suggestions.

Reply:
Close, but `final` also has to be removed for this to compile with your other suggestions.